src/function/aggregate/nested/histogram.cpp

```diff
@@ -0,0 +1,227 @@
+#include "duckdb/function/aggregate/nested_functions.hpp"
+#include "duckdb/planner/expression/bound_aggregate_expression.hpp"
+#include "duckdb/common/pair.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/common/map.hpp"
+#include "duckdb/common/types/vector.hpp"
+#include "duckdb/common/types/chunk_collection.hpp"
+
+namespace duckdb {
+template <class T>
+struct HistogramAggState {
+	map<T, size_t> *hist;
+};
+
+struct HistogramFunction {
+	template <class STATE>
+	static void Initialize(STATE *state) {
+		state->hist = nullptr;
+	}
+
+	template <class STATE>
+	static void Destroy(STATE *state) {
+		if (state->hist) {
+			delete state->hist;
+		}
+	}
+
+	static bool IgnoreNull() {
+		return true;
+	}
+};
+
+template <class T>
+static void HistogramUpdateFunction(Vector inputs[], FunctionData *, idx_t input_count, Vector &state_vector,
+                                    idx_t count) {
+	D_ASSERT(input_count == 1);
+
+	auto &input = inputs[0];
+	VectorData sdata;
+	state_vector.Orrify(count, sdata);
+	VectorData input_data;
+	input.Orrify(count, input_data);
+
+	auto states = (HistogramAggState<T> **)sdata.data;
+	for (idx_t i = 0; i < count; i++) {
+		if (!(*input_data.nullmask)[input_data.sel->get_index(i)]) {
+			auto state = states[sdata.sel->get_index(i)];
+			if (!state->hist) {
+				state->hist = new map<T, size_t>();
+			}
+			T value = input.GetValue(i).GetValue<T>();
```

Comment:
Can we change this to use the underlying array directly (i.e. `auto input_data = (T *) input_data.data;`) instead of the much slower value interface?

Reply:
Sure, I guess I'll have to write a specialized method for strings then